### PR TITLE
Exodii Launcher Sidearms

### PR DESCRIPTION
**Pull request by tenmillimaster**
#### Summary
Content "Adds Long awaited Exodii Sidearms"

#### Purpose of change

The Exodii deserve sidearms. The Exodii, doing their best to adapt to fighting an enemy that does not easily succumb to non-mortal wounds, would have sidearms different to ours. Since our sidearms take on massive penalties to lethality in the name of portability, I sought to cherry pick a technological scenario that could plausibly produce such a firearm as the Exodii would want -- As a rule almost all exodii equipment is not purpose built for the task, after all, but rather adapted. Therefore, like the rifles, the sidearms would boast tremendous lethality, but also stupendous recoil. Perhaps carry multi-purpose utility as well.

#### Describe the solution

On the civilian side:
Punt guns are heavy fowling pieces that are used to take on scores of birds  to bring to market in commercial hunting.
Big game guns are heavy rifles/smoothbores that are used to shoot at dangerous game.
On the martial side:
Swivel guns are light cannon mounted to the deck for shooting at other seamen.
Wall guns are heavy guns/light cannon that are mounted to battlements for shooting at oncoming troops.

So what if you had a nation of avid hunters and fowlers who had to push their civilian firearms into service during a war? You get light swivel/wall guns.  And these are about the right bore diameter for useful payload launching capability, and the actions are plenty strong enough less powerful launching of explosives, whose payloads aren't kinetic energy dependent.

So you chop down a number of these guns for use as launchers and sidearms (howdahs). You get a gun that the humans would have used with the smaller launcher cartridges, but the exodii chamber the original cartridges in.

The numbers for the large cartridge are based on the 4-bore cartridge, a 1 inch bore caliber that was popular pre- expressgun for taking dangerous game and itself evolved from 4-gauge shotguns.

The numbers for the smaller cartridge are based on the the sturmpistole, an adapted flare pistol that was used for launching flares and grenades. An additional rifle grenade of the same origin is also included.

#### Describe alternatives you've considered

Chopped down shotguns, but then these would not have such an extensive array of military loads available.

#### Testing

Ongoing. There's a lot of stuff I need to doublecheck.
#### Additional context

In draft for a bit longer. Need to figure out how to add addn'l  trader levels right now.

### CODE REVIEW

File: data/json/items/ammo/exodii.json [hunk 1]
```diff
@@ -33,5 +33,182 @@
...
+    "dispersion": 30,
+    "range": 30,
+    "damage": { "damage_type": "bullet", "amount": 16 }
+  },
+  {
+    "id": "273x44flare",
+    "copy-from": "273x44abstract",
+    "type": "AMMO",
+    "name": { "str_sp": "27.3x44 Diyua star" },
+    "description": "These flare cartridges are for use in 27.3x44 Diyua guns, or perhaps in 27.3x110 Diyua guns.  They are employed by Exodii for emergency signalling when their usual communications systems are suffering interference, breakdown, or are being jammed.",
```
Comment (TheShadowFerret) [inline data/json/items/ammo/exodii.json:144]:
```suggestion
    "description": "These flare cartridges are for use in 27.3x44 Diyua guns, or perhaps in 27.3x110 Diyua guns.  They are employed by Exodii for emergency signaling when their usual communications systems are suffering interference, breakdown, or are being jammed.",
```

File: data/json/items/ammo/exodii.json [hunk 2]
```diff
@@ -33,5 +33,182 @@
...
+    "price_postapoc": 1800,
+    "damage": { "damage_type": "bullet", "amount": 45 },
+    "effects": [ "COOKOFF" ]
+  },
+  {
+    "id": "273x44thermobaric",
+    "copy-from": "273x44trainer",
+    "type": "AMMO",
+    "name": { "str_sp": "27.3x44 Diyua thermobaric" },
+    "description": "A 27.3x44 Diyua cartridge with a minute amount of liquid fuel encasing a fuel dispersing explosive charge, and a secondary ignition charge.  On impact the fuel is dispersed, creating a fuel air mixture which is ignited a few milliseconds later.  In a cartridge like 27.3x44 Diyua, the pressure wave produced is rather diminuative, yet it makes the most of what little payload is available.",
```
Comment (TheShadowFerret) [inline data/json/items/ammo/exodii.json:173]:
```suggestion
    "description": "A 27.3x44 Diyua cartridge with a minute amount of liquid fuel encasing a fuel dispersing explosive charge, and a secondary ignition charge.  On impact the fuel is dispersed, creating a fuel-air mixture which is ignited a few milliseconds later.  In a cartridge like 27.3x44 Diyua, the pressure wave produced is rather diminutive, yet it makes the most of what little payload is available.",
```

File: data/json/items/ammo_types.json [hunk 3]
```diff
@@ -767,6 +767,24 @@
     "name": "12.3ln cartridge",
     "default": "123ln"
   },
+  {
+    "type": "ammunition_type",
+    "id": "273x110",
+    "name": "2.73x110 Diyua",
```
Comment (TheShadowFerret) [inline data/json/items/ammo_types.json:773]:
```suggestion
    "name": "27.3x110 Diyua",
```

File: data/json/items/ammo_types.json [hunk 4]
```diff
@@ -767,6 +767,24 @@
...
+    "default": "273x110canister"
+  },
+  {
+    "type": "ammunition_type",
+    "id": "273x44",
+    "name": "2.73x44 Diyua",
```
Comment (TheShadowFerret) [inline data/json/items/ammo_types.json:779]:
```suggestion
    "name": "27.3x44 Diyua",
```

File: data/json/items/ammo_types.json [hunk 5]
```diff
@@ -767,6 +767,24 @@
...
+    "default": "273x44HE"
+  },
+  {
+    "type": "ammunition_type",
+    "id": "273x44RG",
+    "name": "2.73x44 Diyua Grenade",
```
Comment (TheShadowFerret) [inline data/json/items/ammo_types.json:785]:
```suggestion
    "name": "27.3x44 Diyua Grenade",
```
No skinny grenades for you

File: data/json/items/generic/casing.json [hunk 6]
```diff
@@ -541,5 +541,25 @@
     "description": "An empty casing from a .30 Carbine round.",
     "weight": "6 g",
     "volume": "8ml"
+  },
+  {
+    "id": "273x110casing",
+    "copy-from": "casing",
+    "looks_like": "50beowulf_casing",
+    "type": "GENERIC",
+    "name": { "str": "27.3 x 110 Diyua casing" },
```
Comment (TheShadowFerret) [inline data/json/items/generic/casing.json:550]:
```suggestion
    "name": { "str": "27.3x110 Diyua casing" },
```

File: data/json/items/generic/casing.json [hunk 7]
```diff
@@ -541,5 +541,25 @@
...
+    "description": "An empty giant brass shotshell, with illegible alien text on the headstamp.  These are rare, so you might want to hold onto them.",
+    "weight": "19 g",
+    "volume": "13ml"
+  },
+  {
+    "id": "273x44casing",
+    "copy-from": "casing",
+    "looks_like": "50beowulf_casing",
+    "type": "GENERIC",
+    "name": { "str": "27.3 x 44 Diyua casing" },
```
Comment (TheShadowFerret) [inline data/json/items/generic/casing.json:560]:
```suggestion
    "name": { "str": "27.3x44 Diyua casing" },
```

File: data/json/items/gun/exodii.json [hunk 8]
```diff
@@ -1,4 +1,124 @@
...
+    "volume": "1775 ml",
+    "longest_side": "428 mm",
+    "price": 850000,
+    "price_postapoc": 2800,
+    "variant_type": "gun",
+    "variants": [
+      {
+        "id": "Brogyeki",
+        "name": { "str": "Brogyeki '23" },
+        "description": "Originating from big game express smoothbores and heavy fowling pieces adapted for wallguns, the Brogyeki '23 is an over/under break action 'pistol' that is (unfortunately?) capable of launching its original 27.3 x 110 Diyua express slug and shot loads as well as the wide array of payloads in 27.3 x 44 Diyua.  A simple wire stock collapses neatly along its octagonal barrels, which feature a combination of sighting devices necessary for launching its numerous loads, in addition to the usual Exodii retrofits.  Unless you are a quadrupedal cyborg with this gun in a shock absorbing mount, recoil with the 27.3 x 110 Diyua shells will be *stout*.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/exodii.json:20]:
```suggestion
        "description": "Originating from big game express smoothbores and heavy fowling pieces adapted for wallguns, the Brogyeki '23 is an over/under break action 'pistol' that is (unfortunately?) capable of launching its original 27.3x110 Diyua express slug and shot loads as well as the wide array of payloads in 27.3x44 Diyua.  A simple wire stock collapses neatly along its octagonal barrels, which feature a combination of sighting devices necessary for launching its numerous loads, in addition to the usual Exodii retrofits.  Unless you are a quadrupedal cyborg with this gun in a shock absorbing mount, recoil with the 27.3x110 Diyua shells will be *stout*.",
```

File: data/json/items/gun/exodii.json [hunk 9]
```diff
@@ -1,4 +1,124 @@
...
+    "volume": "2980 ml",
+    "longest_side": "110 cm",
+    "price": 660000,
+    "price_postapoc": 3000,
+    "variant_type": "gun",
+    "variants": [
+      {
+        "id": "Brogyaga",
+        "name": { "str": "Brogyaga" },
+    "description": "An over/under fowling piece with a long barrel, Brogyaga were favored for their versatility in taking large game and fowl with their 27.3 x 110 Diyua shells.  They would be pressed into combat during two global conflicts, initially as 'light' wall guns, and later as the downsized and down-loaded Brogyeki '23, before being brought along by the Exodii to dimensions unknown.  Though unwieldy in combat as either a primary or secondary, Brogyaga serve as useful parts donors for Brogyeki counterparts, fantastic pieces for the occasional hunting party, and popular trade goods for the locals.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/exodii.json:57]:
```suggestion
         "description": "An over/under fowling piece with a long barrel, Brogyaga were favored for their versatility in taking large game and fowl with their 27.3x110 Diyua shells.  They would be pressed into combat during two global conflicts, initially as 'light' wall guns, and later as the downsized and down-loaded Brogyeki '23, before being brought along by the Exodii to dimensions unknown.  Though unwieldy in combat as either a primary or secondary, Brogyaga serve as useful parts donors for Brogyeki counterparts, fantastic pieces for the occasional hunting party, and popular trade goods for the locals.",
```

File: data/json/items/gun/exodii.json [hunk 10]
```diff
@@ -1,4 +1,124 @@
...
+    "volume": "1930 ml",
+    "longest_side": "980 mm",
+    "price": 1320000,
+    "price_postapoc": 10000,
+    "variant_type": "gun",
+    "variants": [
+      {
+        "id": "Sapra",
+        "name": { "str": "Sapra '36" },
+        "description": "The Sapra '36 is a short smoothbore bolt-action grenade launcher, fitted with a 5-round box magazine and a large, efficent muzzle brake.  Designed in the interwar period following the proliferation of Brogyeki '23, the Sapra was intended to assist in riots, ambushes and assaulting in urban operations, where explosives applied in quick succession could make short work of targets.  To further assist in riot operations, the Sapra can be fitted with a bayonet, a frivolity that the Exodii almost immediately discarded.  Unlike the Brogyeki, the Sapra is incapable of chambering the longer 27.3 x 110 Diyua shells.  The Exodii have replaced the stock with an aluminum alloy chassis, leaving behind the original wood ones as they became damaged.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/exodii.json:94]:
```suggestion
        "description": "The Sapra '36 is a short smoothbore bolt-action grenade launcher, fitted with a 5-round box magazine and a large, efficient muzzle brake.  Designed in the interwar period following the proliferation of Brogyeki '23, the Sapra was intended to assist in riots, ambushes and assaulting in urban operations, where explosives applied in quick succession could make short work of targets.  To further assist in riot operations, the Sapra can be fitted with a bayonet, a frivolity that the Exodii almost immediately discarded.  Unlike the Brogyeki, the Sapra is incapable of chambering the longer 27.3x110 Diyua shells.  The Exodii have replaced the stock with an aluminum alloy chassis, leaving behind the original wood ones as they became damaged.",
```

File: data/json/items/magazine/exodii.json [hunk 11]
```diff
@@ -37,5 +37,24 @@
     "reload_time": 180,
     "flags": [ "MAG_BULKY" ],
     "pocket_data": [ { "pocket_type": "MAGAZINE", "rigid": true, "ammo_restriction": { "123ln": 60 } } ]
+  },
+  {
+    "id": "exodiisapramag5",
+    "looks_like": "exodiiminimag",
+    "type": "MAGAZINE",
+    "name": { "str": "Sapra '36 magazine" },
+    "description": "A large, 5 round single stack magazine for the Sapra '36.",
```
Comment (TheShadowFerret) [inline data/json/items/magazine/exodii.json:46]:
```suggestion
    "description": "A large, five round single stack magazine for the Sapra '36.",
```
Proper gramma' demands <10 is spelt

**Review by TheShadowFerret — COMMENTED** [review]
Not in comments:
1) 27.3x110/44 *whats*? Units good 😛
2) Consistency with vanilla gun/ammo/magazine files is that these new items should be split from 12.3ln(file for which should probably be renamed to 123ln.json)

**Comment by tenmillimaster** [comment]
> Not in comments:
> 
> 1. 27.3x110/44 _whats_? Units good 😛

Where Ln is the unit of measurement for 1.23 Ln, Diyua is the unit of measurement for 27.3x44/110. (27.3 Diyua is about 1 inch, or 25.4mm.)
> 2. Consistency with vanilla gun/ammo/magazine files is that these new items should be split from 12.3ln(file for which should probably be renamed to 123ln.json)
Eh, I thought about this but figure the file was named Exodii for a reason. Looking at the Hub guns, they're all in one file too.

File: data/json/items/ammo/exodii.json [hunk 12]
```diff
@@ -33,5 +33,182 @@
...
+    "damage": { "damage_type": "bullet", "amount": 138, "armor_penetration": 26 },
+    "effects": [ "COOKOFF" ]
+  },
+  {
+    "id": "273x110slugheavy",
+    "copy-from": "273x110abstract",
+    "//": "2624 fps, 2000gr, rngmax:2049yds",
+    "type": "AMMO",
+    "name": { "str_sp": "27.3x110 Diyua heavy slug" },
+    "description": "27.3x110 Diyua slug loads similiar to these were employed by colonists on some alternate earth to take extremely dangerous game.  This heavy slug load however, was used almost entirely by recoil enthusiasts and braggarts, until the Exodii came upon them.",
```
Comment (harakka) [inline data/json/items/ammo/exodii.json:89]:
```suggestion
    "description": "27.3x110 Diyua slug loads similar to these were employed by colonists on some alternate earth to take extremely dangerous game.  This heavy slug load however, was used almost entirely by recoil enthusiasts and braggarts, until the Exodii came upon them.",
```

File: data/json/items/ammo/exodii.json [hunk 13]
```diff
@@ -33,5 +33,182 @@
...
+    "color": "yellow",
+    "count": 1,
+    "stack_size": 1,
+    "ammo_type": "273x44RG",
+    "casing": "273x44casing",
+    "range": 9,
+    "dispersion": 30,
+    "recoil": 22250,
+    "name": { "str_sp": "66/27.3x278 Diyua multipurpose" },
+    "description": "A long 27.3 Diyua cartridge loaded with a propelling charge, attached to a large high explosive shaped charge grenade several times larger in diameter.  It was designed to be muzzle loaded in 27.3x44 Diyua guns.  Though the grenade's penetration and fragmentation were ultimately deemed insufficient, this was of no matter to the Exodii who saw value in the explosive potential the grenade added to the Brogyeki '23.",
```
Comment (harakka) [inline data/json/items/ammo/exodii.json:210]:
```suggestion
    "description": "A long 27.3 Diyua cartridge loaded with a propelling charge, attached to a large high explosive shaped charge grenade several times larger in diameter.  It was designed to be muzzle loaded into 27.3x44 Diyua guns.  Though the grenade's penetration and fragmentation were ultimately deemed insufficient, this was of no matter to the Exodii who saw value in the explosive potential the grenade added to the Brogyeki '23.",
```

**Review by harakka — CHANGES_REQUESTED** [review]

**Comment by tenmillimaster** [comment]
Did the generic guns stuff.

**Pull request merged**